src/rust/engine/src/core.rs

```diff
@@ -164,22 +164,24 @@ pub const ANY_TYPE: TypeId = TypeId(0);
 pub struct Function(pub Key);
 
 impl Function {
-  fn pretty_print(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+  pub fn name(&self) -> String {
     let Function(key) = self;
+    let module = externs::project_str(&externs::val_for(&key), "__module__");
     let name = externs::project_str(&externs::val_for(&key), "__name__");
-    write!(f, "{}()", name)
+    let line_number = externs::project_str(&externs::val_for(&key), "__line_number__");
```

Comment:
Maybe add a comment saying that this is a custom property we set on things, and not a standard python thing (so if it's missing somewhere, it's because we didn't set it, and probably need to)?